### PR TITLE
Build: Cuda without raja 

### PR DESCRIPTION
**Pull request by cnpetra**
CUDA KKT and Matrix classes now use the newly added cuda vector. This removes dependencies of `HIOP_USE_CUDA` on `HIOP_USE_RAJA`. 

Close https://github.com/LLNL/hiop/issues/578

### CODE REVIEW

File: src/LinAlg/hiopMatrixSparseCsrCuda.cpp [hunk 1]
```diff
@@ -301,11 +298,8 @@ void hiopMatrixSparseCSRCUDA::addDiagonal(const double& alpha, const hiopVector&
 {
   assert(nrows_ == D.get_size());
   assert(nrows_ == ncols_ && "Matrix must be square");
-#ifdef HIOP_USE_RAJA
-  assert(dynamic_cast<const hiopVectorCuda*>(&D) && "input vector must be RAJA-CUDA");
-#else
-  assert("input vector must be RAJA-CUDA, but HiOp was not built with RAJA");
-#endif
+  assert(dynamic_cast<const hiopVectorCuda*>(&D) && "input vector must be CUDA");
```
Comment (CameronRutherford) [inline src/LinAlg/hiopMatrixSparseCsrCuda.cpp:301]: I tried to figure this out from other LinAlg objects, but couldn't get that far - should we place any of these asserts behind `#ifdef DEEPCHECKS`, or are they necessary to have during all HiOp execution?

File: src/LinAlg/hiopVectorCuda.cpp [hunk 2]
```diff
@@ -192,6 +194,17 @@ void hiopVectorCuda::copy_from_w_pattern(const hiopVector& vv, const hiopVector&
   componentMult(select);
 }
 
+
+void hiopVectorCuda::copy_from_vec_par(const hiopVectorPar& src)
+{
+  assert(n_local_ == src.get_size());
```
Comment (CameronRutherford) [inline src/LinAlg/hiopVectorCuda.cpp:200]: Same for assertions in this file

File: src/Optimization/hiopKKTLinSysSparseCondensed.cpp [hunk 3]
```diff
@@ -181,24 +170,24 @@ bool hiopKKTLinSysCondensedSparse::build_kkt_matrix(const hiopPDPerturbation& pd
...
   //temporary code, see above note
   {
-    if(mem_space_internal == "DEVICE") {
+    if(mem_space_internal == "CUDA") {
 #ifdef HIOP_USE_CUDA
-      auto Hd_raja = dynamic_cast<hiopVectorCuda*>(Hd_copy_);
+      auto Hd_cuda = dynamic_cast<hiopVectorCuda*>(Hd_copy_);
       auto Hd_par =  dynamic_cast<hiopVectorPar*>(Hd_);
-      assert(Hd_raja && "incorrect type for vector class");
+      assert(Hd_cuda && "incorrect type for vector class");
```
Comment (CameronRutherford) [inline src/Optimization/hiopKKTLinSysSparseCondensed.cpp:177]: Similar for these assertions.

**Review by CameronRutherford — APPROVED** [review]
Just one stylistic comment about assertions in `LinAlg` and `Optimization` code.

Apart from that this looks great!

File: src/LinAlg/hiopMatrixSparseCsrCuda.cpp (hunk 1 above)
Comment (cnpetra) [reply src/LinAlg/hiopMatrixSparseCsrCuda.cpp:301]: they catch a supposedly "no-go" case caused by a potential change in the code by a developer (use of improper HiOp vector type), so it should pop up under debug builds. It is not user-related nor potentially some abnormal numerics, which sometimes go under deepchecks.

**Comment by cnpetra** [comment]
> LGTM. It seems condensed system and normal equation depends on 'CUDA' vector, doesn't it? May be we should add some comments about this in the user guide and option file (hiopOptions.cpp).

https://github.com/LLNL/hiop/issues/575 is supposed to take care of it

**Pull request merged**